### PR TITLE
Set Composer dependencies for 2.3 branch

### PR DESCRIPTION
**Pull request by mnocon**
Adjusting composer.json so that dependencies from master are not installed.

I've installed v2.3.2 tag using PHP 7.1, it installs (EzSystems packages only):
```
ezsystems/behat-screenshot-image-driver-cloudinary v1.0.0     Cloudinary image driver for bex/behat-scree...
ezsystems/behatbundle                              v6.5.3     Behat bundle for help testing eZ Bundles an...
ezsystems/ez-support-tools                         v0.2.1     Providing information about the system eZ P...
ezsystems/ezplatform-admin-ui                      v1.3.2     eZ Platform Admin v2
ezsystems/ezplatform-admin-ui-assets               v3.1.0     External assets dependencies for EzPlatform...
ezsystems/ezplatform-admin-ui-modules              v1.3.1     External assets dependencies for EzPlatform...
ezsystems/ezplatform-cron                          v2.0.0     Simple cron bundle for eZ Platform
ezsystems/ezplatform-design-engine                 v2.0.0     Design fallback mechanism for eZ Platform
ezsystems/ezplatform-http-cache                    v0.8.0     HTTP cache handling for eZ Platform.
ezsystems/ezplatform-solr-search-engine            v1.5.4     Solr search engine implementation for eZ Pl...
ezsystems/ezplatform-standard-design               v0.1.0     eZ Platform Standard Design Bundle
ezsystems/ezpublish-kernel                         v7.3.2     Kernel used by ezsystems/ezplatform and der...
ezsystems/repository-forms                         v2.3.2     Use Symfony forms with eZ Platform repo
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -35,17 +35,17 @@
...
         "sensio/distribution-bundle": "^5.0.22",
         "incenteev/composer-parameter-handler": "^2.1.3",
-        "ezsystems/ezpublish-kernel": "^7.3@dev",
-        "ezsystems/repository-forms": "^2.3@dev",
-        "ezsystems/ezplatform-solr-search-engine": "^1.5@dev",
-        "ezsystems/ez-support-tools": "^0.2@dev",
-        "ezsystems/ezplatform-http-cache": "^0.8@dev",
-        "ezsystems/ezplatform-admin-ui": "^1.3@dev",
-        "ezsystems/ezplatform-admin-ui-modules": "^1.3@dev",
+        "ezsystems/ezpublish-kernel": "~7.3.2@dev",
```
Comment (micszo) [inline composer.json]: Specifying 3rd digit is a change of approach, could you elaborate? 🙂

Comment (mnocon) [reply composer.json]: How is this a change of approach? The same was done for previous branches (2.2: https://github.com/ezsystems/ezplatform/blob/2.2/composer.json#L38
2.1: https://github.com/ezsystems/ezplatform/blob/2.1/composer.json#L38 ).

The idea is to limit Kernel versions to the 7.3.* line, with 7.3.2 being the lowest version that can be installed - that's why the third digit has to be specified, as ~7.3 means 7.* line (with 7.3 being the lowest versions) - according to https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

Comment (micszo) [reply composer.json]: What I was aiming for was _seems like_. 😉 
OK, thanks, that's the context I was missing.

**Comment by andrerom** [comment]
Please don't merge more changes toi composer.json, I have enough work for the weekend already merging changes from the 1.7 change.

Most of these changes was in what I was about to push anyway.

**Comment by lserwatka** [comment]
@andrerom please communicate properly that you are working on it as nobody is seating in your head. Marek had time so he started. He needs to merge it up.

**Pull request merged**